analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SpecifyTimeoutOnRegex.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+class Compliant
+{
+    void Ctor()
+    {
+        var defaultOrder = new Regex("some pattern", RegexOptions.None, TimeSpan.FromSeconds(1)); // Compliant
+
+        var namedArgs = new Regex(
+            pattern: "some pattern",
+            matchTimeout: TimeSpan.FromSeconds(1), // Compliant
+            options: RegexOptions.None);
```

Comment:
Please use the arguments in an order different from the positional ordering (e.g. inverse the ordering).